pkgs/by-name/so/sobjectizer/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  withStatic ? true,
+  withShared ? true,
+  buildExamples ? false,
+  buildTests ? true,
+  buildAll ? false,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sobjectizer";
+  version = "5.8.4";
+
+  src = fetchFromGitHub {
+    owner = "Stiffstream";
+    repo = "sobjectizer";
+    tag = "v.${finalAttrs.version}";
+    hash = "sha256-tIqWgd6TppHfqZk3XHzhG0t+Nn8BQCTP81UD7ls67UE=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeDir = "../dev";
+
+  cmakeFlags = [
+    (lib.cmakeBool "SOBJECTIZER_BUILD_STATIC" withStatic)
+    (lib.cmakeBool "SOBJECTIZER_BUILD_SHARED" withShared)
+    (lib.cmakeBool "BUILD_EXAMPLES" buildExamples)
+    (lib.cmakeBool "BUILD_TESTS" buildTests)
+    (lib.cmakeBool "BUILD_ALL" buildAll)
+  ];
+
+  # static and shared are both required for tests building.
+  doCheck = withStatic && withShared;
+
+  meta = with lib; {
```

Comment:
Let's move the `with lib;` around single expression

Reply:
I believe this is a non-blocking nit, i.e. the PR can be merged even if you dont do this.

Reply:
> Let's move the `with lib;` around single expression

Done.